File: 665-non-decreasing-array/665-non-decreasing-array.py
class Solution:
    def checkPossibility(self, nums: List[int]) -> bool:
        origin = nums.copy()
        front = True
        back = True
        cnt = 0
        for idx , item in enumerate(nums):
            if front == False:
                break
            if idx == 0 or nums[idx] >= nums[idx-1]:
                continue
            if cnt == 1:
                front = False
                continue
            nums[idx] = nums[idx-1]
            cnt += 1
        cnt = 0
        nums = origin
        for idx in range(len(nums)-2,-1,-1):
            if back == False:
                break
            if nums[idx] <= nums[idx+1]:
                continue
            if cnt == 1:
                print(idx)
                back = False
                continue
            nums[idx] = nums[idx+1]
            cnt += 1
        print(front,back)
        return front or back
        

File: 665-non-decreasing-array/665-non-decreasing-array.java
class Solution {
    public boolean checkPossibility(int[] nums) {
        int[] nums2 = new int[nums.length];
        for (int i = 0 ; i < nums.length ; i ++){
            nums2[i] = nums[i];
        }
        boolean front = true;
        boolean back = true;
        int cnt = 0;
        for (int i = 1 ; i < nums.length ; i++){
            // if (front == false)
            //     break;
            if (nums[i] >= nums[i-1])
                continue;
            if (cnt == 1){
                front = false;
                break;
            }
            nums[i] = nums[i-1];
            cnt ++;
        }
        cnt = 0 ; nums = nums2;
        for (int i = nums.length - 2 ; i >= 0; i--){
            // if (front == false)
            //     break;
            if (nums[i] <= nums[i+1])
                continue;
            if (cnt == 1){
                back = false;
                break;
            }
            nums[i] = nums[i+1];
            cnt ++;
        }
        return front || back;
    }
}